### PR TITLE
Update atomic and non-atomic save handling for removing `type/`

### PR DESCRIPTION
**Pull request by carmenlau**

### CODE REVIEW

File: features/126-record-id/implementation.md [hunk 1]
```diff
@@ -99,40 +99,84 @@ The Record fetch and delete APIs are changed substantially:
...
                    recordIDs:(NSArray<NSString *> *)recordIDs
-           completionHandler:(void (^)(NSDictionary<NSString *, SKYRecord *> *recordsByRecordID,
-                                      NSError *operationError))completionHandler
-      perRecordErrorHandler:(void (^)(NSString *recordID,
-                                      NSError *error))errorHandler;
+           completionHandler:(void (^)(NSArray<id> *records,
+                                       NSArray<id> *operationError))completionHandler NS_REFINED_FOR_SWIFT;
+
+extension SKYDatabase {
+    func fetchRecordsWithType(recordType: NSString,
```
Comment (cheungpat) [inline features/126-record-id/implementation.md]: should be `fetchRecords(type:recordIDs:completion)`

File: features/126-record-id/implementation.md [hunk 2]
```diff
@@ -99,40 +99,84 @@ The Record fetch and delete APIs are changed substantially:
 // New, no longer support fetching multiple types
 - (void)fetchRecordsWithType:(NSString *)recordType
                    recordIDs:(NSArray<NSString *> *)recordIDs
-           completionHandler:(void (^)(NSDictionary<NSString *, SKYRecord *> *recordsByRecordID,
-                                      NSError *operationError))completionHandler
-      perRecordErrorHandler:(void (^)(NSString *recordID,
-                                      NSError *error))errorHandler;
+           completionHandler:(void (^)(NSArray<id> *records,
```
Comment (cheungpat) [inline features/126-record-id/implementation.md]: will be `completion:`

File: features/126-record-id/implementation.md [hunk 3]
```diff
@@ -99,40 +99,84 @@ The Record fetch and delete APIs are changed substantially:
...
+                                       NSArray<id> *operationError))completionHandler NS_REFINED_FOR_SWIFT;
+
+extension SKYDatabase {
+    func fetchRecordsWithType(recordType: NSString,
+                              recordIDs: [NSString],
+                              completionHandler: (records: [SKYRecord?], errors: [NSError?]) -> Void) {
+        ...
```
Comment (cheungpat) [inline features/126-record-id/implementation.md]: syntax error: unexpected `...`

no just kidding : )

File: features/126-record-id/implementation.md [hunk 4]
```diff
@@ -99,40 +99,84 @@ The Record fetch and delete APIs are changed substantially:
...
+        (void (^_Nullable)(SKYRecord *_Nullable record, NSError *_Nullable error))errorHandler;
+
+- (void)saveRecordsAtomically:(NSArray<SKYRecord *> *)records
+            completionHandler:
+                (void (^_Nullable)(NSArray *_Nullable savedRecords,
+                                   NSError *_Nullable operationError))completionHandler;
+
+// New, perRecordErrorHandler will be removed
+- (void)saveRecords:(NSArray<SKYRecord *> *)records
+  completionHandler:(void (^_Nullable)(NSArray<id> *savedRecords,
```
Comment (cheungpat) [inline features/126-record-id/implementation.md]: `completion:`

File: features/126-record-id/implementation.md [hunk 5]
```diff
@@ -99,40 +99,84 @@ The Record fetch and delete APIs are changed substantially:
...
+                                       NSArray<id> *errors))completionHandler NS_REFINED_FOR_SWIFT;
+
+extension SKYDatabase {
+    func saveRecords(records: [SKYRecord],
```
Comment (cheungpat) [inline features/126-record-id/implementation.md]: `saveRecords(_ records: [SKYRecord] completion:)`

File: features/126-record-id/implementation.md [hunk 6]
```diff
@@ -99,40 +99,84 @@ The Record fetch and delete APIs are changed substantially:
...
-           completionHandler:(void (^)(NSArray<NSString *> *deletedRecordIDs,
-                                       NSError *error))completionHandler
-       perRecordErrorHandler:(void (^)(NSString *recordID,
-                                       NSError *error))errorHandler;
-```
+           completionHandler:(void (^)(NSArray<id> *deletedRecordIDs,
+                                       NSArray<id> *errors))completionHandler NS_REFINED_FOR_SWIFT;
+
+extension SKYDatabase {
+    func deleteRecordsWithType(recordType: NSString,
```
Comment (cheungpat) [inline features/126-record-id/implementation.md]: `deleteRecords(type:recordIDs:completion)`

File: features/126-record-id/implementation.md [hunk 7]
```diff
@@ -200,6 +244,39 @@ The SDK does not expose deprecated ID in the API. The Android SDK does not
...
 
+After updating the id format, ids are not necessarily unique in single save
+request. So partially save handler will change from map to array. 
+
+```java
+// old
+public abstract class RecordSaveResponseHandler implements ResponseHandler {
+
+    ...
+    public abstract void onPartiallySaveSuccess(Map<String, Record> successRecords, Map<String, Error> errors);
```
Comment (cheungpat) [inline features/126-record-id/implementation.md]: s/PartiallySave/PartialSave/

File: features/126-record-id/implementation.md [hunk 8]
```diff
@@ -200,6 +244,39 @@ The SDK does not expose deprecated ID in the API. The Android SDK does not
...
+}
+
+public abstract class RecordDeleteResponseHandler implements ResponseHandler {
+
+    ...
+    public abstract void onPartiallySaveSuccess(Map<String, Record> successRecords, Map<String, Error> errors);
```
Comment (ben181231) [inline features/126-record-id/implementation.md]: This should be `onDeletePartialSuccess`

**Comment by carmenlau** [comment]
Updated! Thanks @cheungpat @ben181231 
If it looks good, i will merge it first and the detail record api spec we can update in the api consistency PRs.

**Comment by carmenlau** [comment]
The change of the PR is out of date, for the detail record api interface please reference to the api consistency spec.

**Pull request closed without merging**